### PR TITLE
DEPR: Error with ambiguous groupby strings

### PR DESCRIPTION
**Pull request by gfyoung**
Title is self-explanatory.

xref #14432.

### CODE REVIEW

File: pandas/tests/frame/test_sort_values_level_as_str.py [hunk 1]
```diff
@@ -93,34 +93,3 @@ def test_sort_column_level_and_index_label(
...
-    # Sorting by 'idx' should sort by the idx column and raise a
-    # FutureWarning
-    with tm.assert_produces_warning(FutureWarning):
-        result = df.sort_values(by='idx')
-
-    # This should be equivalent to sorting by the 'idx' index level in
-    # descending order
-    expected = df.sort_index(level='idx', ascending=False)
-    assert_frame_equal(result, expected)
-
```
Comment (jreback) [inline pandas/tests/frame/test_sort_values_level_as_str.py]: are these sufficiently covered as testing that they raise?

File: pandas/tests/groupby/test_index_as_string.py [hunk 2]
```diff
@@ -67,50 +66,3 @@ def test_grouper_index_level_as_string_series(series, levels):
...
-@pytest.mark.parametrize('key_strs,key_groupers,level_groupers', [
-    ('inner',  # Index name
-     pd.Grouper(key='inner'),
-     pd.Grouper(level='inner'),
-     ),
-    (['inner'],  # List of index name
-     [pd.Grouper(key='inner')],
-     [pd.Grouper(level='inner')]
-     ),
-    (['B', 'inner'],  # Column and index
```
Comment (jreback) [inline pandas/tests/groupby/test_index_as_string.py]: maybe just change to have these check for raising? or too duplicative?

File: pandas/tests/reshape/test_pivot.py [hunk 3]
```diff
@@ -596,52 +596,6 @@ def _check_output(result, values_col, index=['A', 'B'],
...
-
-        df = df.set_index(['JOB', 'NAME', 'YEAR', 'MONTH'], drop=False,
-                          append=False)
-
-        with tm.assert_produces_warning(FutureWarning, check_stacklevel=False):
-            result = df.pivot_table(index=['JOB', 'NAME'],
-                                    columns=['YEAR', 'MONTH'],
-                                    values=['DAYS', 'SALARY'],
-                                    aggfunc={'DAYS': 'mean', 'SALARY': 'sum'},
-                                    margins=True)
```
Comment (jreback) [inline pandas/tests/reshape/test_pivot.py]: do these raise now?

File: pandas/tests/reshape/merge/test_merge_index_as_string.py [hunk 4]
```diff
@@ -176,38 +175,3 @@ def test_join_indexes_and_columns_on(df1, df2, left_index, join_type):
...
-    # Construct left_df with both an index and a column named 'outer'.
-    # We make this 'outer' column equal to the 'inner' column so that we
-    # can verify that the correct values are used by the merge operation
-    left_df = df1.set_index('outer')
-    left_df['outer'] = left_df['inner']
-
-    # Construct right_df with an index level named 'outer'
-    right_df = df2.set_index('outer')
-
-    # Construct expected result.
```
Comment (jreback) [inline pandas/tests/reshape/merge/test_merge_index_as_string.py]: do these raise now?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/test_sort_values_level_as_str.py (hunk 1 above)
Comment (gfyoung) [reply pandas/tests/frame/test_sort_values_level_as_str.py]: Raise.

File: pandas/tests/groupby/test_index_as_string.py (hunk 2 above)
Comment (gfyoung) [reply pandas/tests/groupby/test_index_as_string.py]: Too duplicative --> it raises.

File: pandas/tests/reshape/test_pivot.py (hunk 3 above)
Comment (gfyoung) [reply pandas/tests/reshape/test_pivot.py]: Yes.

File: pandas/tests/reshape/merge/test_merge_index_as_string.py (hunk 4 above)
Comment (gfyoung) [reply pandas/tests/reshape/merge/test_merge_index_as_string.py]: Yes.

**Comment by gfyoung** [comment]
@jreback : Any further thoughts on this PR?  I answered all of your questions.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. @jorisvandenbossche @TomAugspurger

**Comment by TomAugspurger** [comment]
Running dask's test suite on this branch quick. I suspect this will break some things.

**Comment by jreback** [comment]
@TomAugspurger if might make sense to expand the test_dask test that we have (which basically just tests imports).

**Comment by TomAugspurger** [comment]
Perhaps, but this would have been a minor edge case.

Things seem good on dask's end, +1

**Comment by jreback** [comment]
thanks @gfyoung

File: doc/source/groupby.rst [hunk 5]
```diff
@@ -106,9 +106,8 @@ consider the following ``DataFrame``:
    .. versionadded:: 0.20
 
    A string passed to ``groupby`` may refer to either a column or an index level.
-   If a string matches both a column name and an index level name then a warning is
-   issued and the column takes precedence. This will result in an ambiguity error
-   in a future version.
+   If a string matches both a column name and an index level name, a
+   ``ValueError`` will be raised.
```
Comment (jorisvandenbossche) [inline doc/source/groupby.rst]: This should no longer be in a versionadded directive?

Comment (gfyoung) [reply doc/source/groupby.rst]: Do we drop `..versionadded` directives from docstrings?

Comment (gfyoung) [reply doc/source/groupby.rst]: I might just refer to drop it, since it really isn't an add either.  Happy to debate semantics 🙂

Comment (gfyoung) [reply doc/source/groupby.rst]: Handled in #22503.

**Pull request merged**